src/io/flutter/bazel/PluginConfig.java

```diff
@@ -15,21 +17,44 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.io.InputStreamReader;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
 
 /**
  * An in-memory snapshot of the flutter.json file from a Bazel workspace.
  */
 public class PluginConfig {
   private final @NotNull Fields fields;
+  private final @Nullable List<Pattern> directoryPatterns;
 
-  private PluginConfig(@NotNull Fields fields) {
+  private PluginConfig(@NotNull Fields fields, @Nullable List<Pattern> patterns) {
     this.fields = fields;
+    this.directoryPatterns = patterns;
   }
 
-  public @Nullable String getFlutterDaemonScript() {
-    return fields.flutterDaemonScript;
+  /**
+   * Returns true if the given path is within a flutter project in this workspace.
+   *
+   * <p>The path should be relative to the workspace root.
+   */
+  public boolean withinFlutterDirectory(String path) {
+    if (directoryPatterns == null) {
+      // Default if unconfigured.
+      return path.contains("flutter");
+    }
+
+    for (Pattern p : directoryPatterns) {
```

Comment:
It's not checked in yet, I'm thinking of having at least "flutter" and "mobile".